src/object/schema_system_catalog_install.cpp

```diff
@@ -1219,120 +1237,830 @@ namespace cubschema
+    {CT_METHFILE_NAME, system_catalog_initializer::init_meth_file ()},
+    {CT_QUERYSPEC_NAME, system_catalog_initializer::init_query_spec ()},
+    {CT_INDEX_NAME, system_catalog_initializer::init_index ()},
+    {CT_INDEXKEY_NAME, system_catalog_initializer::init_index_key ()},
+    {CT_CLASSAUTH_NAME, system_catalog_initializer::init_class_authorization ()},
+    {CT_PARTITION_NAME, system_catalog_initializer::init_partition()},
+    {CT_DATATYPE_NAME, system_catalog_initializer::init_data_type()},
+    {CT_STORED_PROC_NAME, system_catalog_initializer::init_stored_procedure()},
+    {CT_STORED_PROC_ARGS_NAME, system_catalog_initializer::init_stored_procedure_arguments()},
+    {CT_SERIAL_NAME, system_catalog_initializer::init_serial()},
+    {CT_HA_APPLY_INFO_NAME, system_catalog_initializer::init_ha_apply_info()},
+    {CT_COLLATION_NAME, system_catalog_initializer::init_collations()},
+    {CT_CHARSET_NAME, system_catalog_initializer::init_charsets()},
+    {CT_DUAL_NAME, system_catalog_initializer::init_dual()},
+    {CT_SYNONYM_NAME, system_catalog_initializer::init_synonym()},
+    {CT_DB_SERVER_NAME, system_catalog_initializer::init_db_server()}
+  };
+
+  static std::vector<cubschema::catcls_function_ng> vcl =
+  {
+    {CTV_CLASS_NAME, system_catalog_initializer::init_view_class ()},
+    {CTV_SUPER_CLASS_NAME, system_catalog_initializer::init_view_super_class ()},
+    {CTV_VCLASS_NAME, system_catalog_initializer::init_view_vclass ()},
+    {CTV_ATTRIBUTE_NAME, system_catalog_initializer::init_view_attribute ()},
+    {CTV_ATTR_SD_NAME, system_catalog_initializer::init_view_attribute_set_domain ()},
+    {CTV_METHOD_NAME, system_catalog_initializer::init_view_method ()},
+    {CTV_METHARG_NAME, system_catalog_initializer::init_view_method_argument ()},
+    {CTV_METHARG_SD_NAME, system_catalog_initializer::init_view_method_argument_set_domain ()},
+    {CTV_METHFILE_NAME, system_catalog_initializer::init_view_method_file ()},
+    {CTV_INDEX_NAME, system_catalog_initializer::init_view_index ()},
+    {CTV_INDEXKEY_NAME, system_catalog_initializer::init_view_index_key ()},
+    {CTV_AUTH_NAME, system_catalog_initializer::init_view_authorization ()},
+    {CTV_TRIGGER_NAME, system_catalog_initializer::init_view_trigger ()},
+    {CTV_PARTITION_NAME, system_catalog_initializer::init_view_partition ()},
+    {CTV_STORED_PROC_NAME, system_catalog_initializer::init_view_stored_procedure ()},
+    {CTV_STORED_PROC_ARGS_NAME, system_catalog_initializer::init_view_stored_procedure_arguments ()},
+    {CTV_DB_COLLATION_NAME, system_catalog_initializer::init_view_db_collation ()},
+    {CTV_DB_CHARSET_NAME, system_catalog_initializer::init_view_db_charset ()},
+    {CTV_SYNONYM_NAME, system_catalog_initializer::init_view_synonym ()},
+    {CTV_DB_SERVER_NAME, system_catalog_initializer::init_view_db_server ()}
+  };
+
+  clist = &cl;
+  vclist = &vcl;
+}
+
+int
+catcls_install_class (void)
```

Comment:
It seems possible to unify catcls_install_class and catcls_install_vclass with a flag that indicates which build function should be called, build_class and build_vclass.